### ISSUE TITLE
some issues about db connection

### ISSUE THREAD

**Issue by lwz7512** [ttymck/fastapi-fullstack-boilerplate#1]
Hi ttymck, thank you for sharing your project! It's great!

I ran into few issues in calling `/api`, after some research I fixed.  I would like to share my fix:

for `.env.example`, the DB_URL should be:

```
DB_URL='postgresql+psycopg2://${DB_HOST}/${DB_NAME}'
```

for `dependencies.py`, there should have a new function for `APIRoutes` class:

```
def get_db():
    with get_db_session() as db:
        return db
```

in `main.py`,  APIRoutes depends should be:

```
from .dependencies import get_db, Templates, Render
db: Session = Depends(get_db)
```

thanks for your  sharing!

**Comment by tmkontra** [comment]
Thanks @lwz7512 ! Thanks for catching these mistakes. I've pushed a patch that should resolve them.

To clarify around the `get_db_session` "dependency": I had previously used a dependency like one you have suggested. It usually works pretty well! However, it can unfortunately lead to unexpected behavior around session closure and possibly lead to deadlock. EDIT: I meant to say that I changed the codebase at one point, switching from the "dependency" to the raw context manager. That's why the invalid `Depends(get_db_session)` was present. I have kept it to illustrate both methods of db session creation.

I added the dependency version of `get_db` like you suggested, with a warning and a link to the github discussion about the possible issues. I personally recommend creating a db session within the body of each route.
